### DISCUSSION TITLE
Finalizing setup for the Essentials/Wesmartify and Ucomen power strips

### DISCUSSION THREAD

**Discussion by karlitos** [arendst/Tasmota#16040] [Q&A] [answered]
Hello community,

I bought one Essentials Smart Home 3-socket USB Power Strip from [Amazon](https://www.amazon.de/gp/product/B09PVG364V) which is sometimes also known under the Wesmartify brand. It is almost identical with an [Ucomen branded power strip](https://www.amazon.de/gp/product/B08RJ3MN76/) in black color.

The Essentials power strip is powered with an Tywe2s module which is pretty good accessible. So flashing Tasmota with an serial adapter was no problem once I removed a bunch of [one-way security screws](https://de.wukihow.com/wiki/Remove-One-Way-Screws).  I already figured out the GPIO assignment for the 3 relays and the one switch and I'd love to add those devices to the supported device list, but I still have some minor issues:

The template looks like this:
```
{"NAME":"Essentials Smart Home 3-socket USB Power Strip","GPIO":[0,0,0,0,544,226,0,0,224,32,225,0,0,0],"FLAG":0,"BASE":18}
```

 and setting the Button to 1,2 or 3 respectively toggles the relays. Only sometimes it take multiple presses, till Tasmota realizes the button press, or I need to toggle some relay in the the UI - afterwards, it toggles the related relay quite reliable. Is there anything to do to improve the setup ?

Also, after pressing the button, the LinkLed flashes shortly every time - I do not know, if this is right or wrong.

In the original Firmware - the button toggles all 3 sockets at one and I would like to have it lke this with tasmota. Did not figure out how to do this.

![IMG_20220720_153135](https://user-images.githubusercontent.com/3693437/180244310-347ea1dd-02c4-4533-a63f-201795c556d1.jpg)
![IMG_20220720_153601](https://user-images.githubusercontent.com/3693437/180244373-3d9fa1ca-c9d7-4516-ac0a-94e0eb7051cd.jpg)
![IMG_20220720_153510](https://user-images.githubusercontent.com/3693437/180244345-a6b2153e-c415-42a8-8fe0-792e39e48c94.jpg)

**Comment by karlitos** [comment]
I just read the wiki and figured out following commands:

```
SetOption73 1
Rule1 1
Rule1 on button1#state do Power0 2 endon
```

This works quite well so far, but it toggles the states of the button, so I still need to figure out how to set some variables/counters, so that the first button press would turn OFF all relays, next turn ON all ... next turn OFF all ... and so on

**Reply by sfromis** [reply]
The button has no remembered state which could be toggled. Or do you mean quick presses? Such could be detected as multi-press, unless you use SetOption13 to config for immediate action on single button press.

**Reply by karlitos** [reply]
I am reading about variables right now. Trying to figure out how to do the NOT operator. Then, the rule might look be like this:
```
var1=0
Rule1 on button1#state do backlog  Power0 %var1%; var1=NOT%var1% endon
```
There is some code example on [toggling variables](https://tasmota.github.io/docs/Rules/#setting-variables) in the Wiki, I am not sure if it is the right way

**Reply by sfromis** [reply]
There is no `NOT` operator, that would have been included in the docs.

The `Scale` command can be used to flip between values 0 and 1.
`Scale1 %var1%,1,0,0,1`
The output includes decimals. While this does not break the result of the `Power` command, you can config with `CalcRes 0` to have "clean" values.

**Comment by karlitos** [comment, marked as answer]
I solved the switch all on/off with the single/double click button state. Works great.
```
SetOption73 1
Rule1 1
Rule1 on button1#state=10 do Power0 0 endon
Rule2 1
Rule2 on button1#state=11 do Power0 1 endon
```
after playing with the power strip for some hours is feels very stable, the button assignment works after reboot without problem, I do not know whether I should worry about the blinking LedLInk when the button is pressed, but I did not notice any negative side effects so far.

So with this, I will add this device to the supported device list

**Reply by sfromis** [reply]
Standard setup includes LedLink blink with Wifi activity, unless you disable that led usage. No need to worry, unless it becomes inconvenient.
